### PR TITLE
dolt sql-server bats test support

### PR DESCRIPTION
**Pull request by bheni**

### CODE REVIEW

**Comment by bheni** [comment]
@coffeegoddd I'm seeing this message on Jenkins:

```not ok 365 run sql server and verify connection
# (from function `wait_for_connection' in file ./helper/query-server-common.bash, line 3,
#  from function `setup' in test file ./sql-server.bats, line 10)
#   `wait_for_connection 5000' failed
# Successfully initialized dolt data repository.
# Starting server on port 6095.Traceback (most recent call last):
#   File "<string>", line 9, in <module>
#   File "/home/jenkins/agent/workspace/liquidata-inc_dolt_PR-469/bats/helper/pytest.py", line 4, in <module>
#     import mysql.connector
# ImportError: No module named 'mysql'
# /home/jenkins/agent/workspace/liquidata-inc_dolt_PR-469/bats/./helper/common.bash: line 49:  5108 Terminated              dolt sql-server --port=$SQL_PORT```

I believe doltpy is supposed to be installed.  doltpy has the python mysql connector as a dependency.

Homepage
https://dev.mysql.com/doc/connector-python/en/connector-python-introduction.html

Installation
```pip3 install mysql-connector-python```

Would appreciate some help figuring out why this dependency didn't get installed.

File: bats/helper/query-server-common.bash [hunk 1]
```diff
@@ -0,0 +1,75 @@
+
+wait_for_connection() {
+    PYTEST_DIR=`batshelper`
```
Comment (Hydrocharged) [inline bats/helper/query-server-common.bash]:
```suggestion
    PYTEST_DIR="$BATS_TEST_DIRNAME/helper"
```

File: bats/helper/query-server-common.bash [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+# server_query connects to a running mysql server, executes a query and compares the results against what is expected.
+# In the event that the results do not match expectations, the python process will exit with an exit code of 1
+#  * param1 is the query_str
+#  * param2 is a csv representing the expected result set.  If a query is not expected to have a result set "" should
+#      be passed.
+server_query() {
+    let PORT="$$ % (65536-1024) + 1024"
+    PYTEST_DIR=`batshelper`
```
Comment (Hydrocharged) [inline bats/helper/query-server-common.bash]:
```suggestion
    PYTEST_DIR="$BATS_TEST_DIRNAME/helper"
```

File: bats/.gitignore [hunk 3]
```diff
@@ -0,0 +1 @@
+helper/__pycache__
```
Comment (bheni) [inline bats/.gitignore]: running python from bats creates this __pycache__ directory which we don't want checked in.

**Comment by bheni** [comment]
@andrew-wm-arthur Can you review the python in bats/helper/pytest.py and bats/helper/query-server-common.bash?
@timsehn can you review the non-python stuff?

**Comment by bheni** [comment]
Wasn't really trying to test exhaustively, just wanted to get in a framework for dolt sql-server testing in bats.  Can follow up with more thorough test cases.

**Comment by timsehn** [comment]
I'm not even sure if alter and drop work in the SQL Server. I'm good with
checking it in.

**Pull request merged**